addons/mass_mailing/models/mass_mailing.py

```diff
@@ -57,6 +59,67 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def action_merge(self, src_lists, archive):
```

Comment:
Here add self to src instead of waiting people to put their dst into src manually.